rasgoql/rasgoql/data/sqlalchemy.py

```diff
@@ -150,32 +61,28 @@ def change_namespace(self, namespace: str) -> None:
         `namespace`: str:
             namespace (database.schema)
         """
-        namespace = self.validate_namespace(namespace)
-        database, schema = self.parse_namespace(namespace)
-        try:
-            self.execute_query(f"USE DATABASE {database}")
-            self.execute_query(f"USE SCHEMA {schema}")
-            self.default_namespace = namespace
-            self.default_database = database
-            self.default_schema = schema
-            verbose_message(f"Namespace reset to {self.default_namespace}", logger)
-        except Exception as e:
-            self._error_handler(e)
+        raise NotImplementedError(
+            "Connecting to a new Database in a single session is not supported "
+            "by the SQLAlchemy Engine. Please build a new connection."
+        )
+
 
     @abstractmethod
-    def connect(self, credentials: dict):
+    def connect(self, credentials: Union[dict, DWCredentials]):
         """
         Connect to DB
         Params:
         `credentials`: dict:
             dict  holding the connection credentials
         """
+        if isinstance(credentials, DWCredentials):
+            credentials = credentials.to_dict()
+
         try:
             self.credentials = credentials
             self.database = credentials.get("database")
             self.schema = credentials.get("schema")
-            self.connection = alchemy_session(self._engine)
-            verbose_message("Connected to DB", logger)
```

Comment:
Removing `verbose_message` as it cannot be used here under normal workflow conditions anyways. Due to the connect needing to happen to return this object before `set_verbose` can be called.